runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/StreamingDataflowWorker.java

```diff
@@ -2027,6 +2034,10 @@ private void updateVMMetrics() {
   private void updateThreadMetrics() {
     timeAtMaxActiveThreads.getAndReset();
     timeAtMaxActiveThreads.addValue(workUnitExecutor.allThreadsActiveTime());
+    activeThreads.getAndReset();
+    activeThreads.addValue(workUnitExecutor.activeCount());
+    maxActiveThreads.getAndReset();
+    maxActiveThreads.addValue(chooseMaximumNumberOfThreads());
```

Comment:
If you make above `intMax`, I believe here we should use `workUnitExecutor.activeCount()` as well?

Otherwise it's just a different metric for `numberOfWorkerHarnessThreads`? Not actual max utilized

Reply:
We are ok if it is the same as numberOfWorkerHarnessThreads, we just want to have it exported as a counter, in case we dynamically adjust thread limit later on

Reply:
This also doesn't make sense to me. If `numberOfWorkerHarnessThreads` is good enough, why don't we just use that? If we eventually decide to adjust the thread limit, can't we react at that point?

Reply:
The counter name maxActiveThreads is misleading. Instead, I believe the intention is to measure the number of total allocated threads for the worker. We should probably rename the counter to something along the lines of totalAllocatedThreads.

In this case (maybe I'm wrong here) but I think the counter type (intSum) and value (chooseMaximumNumberOfThreads()) are actually what we want? The end goal is to be able to calculate the ratio between the total # of active threads across all user workers / the total # of threads that are allocated.

Reply:
I see. Yeah it's a bit confusing/misleading, but I guess you are just replicating what we have for javaHarnessUsedMemory/javaHarnessMaxMemory?

Giving it's cumulative, I feel like using a long makes more sense than int. For the default case of 300 threads, we could only collect a limited number of times before overflowing.

Today the default interval (`windmillHarnessUpdateReportingPeriod`) is 10s, so that would blow up after 2147483647/300/(60*60/10*24*365) = ~2 years.

Reply:
Yeah, it's exactly in the same spirit as javaHarnessUsedMemory/javaHarnessMaxMemory.

As for the long vs int counter type, I'm not familiar with how counters are reported, but I was somehow under the impression that with the GetAndReset, the value of this counter would never approach the int max limit? In that way, it's almost acting as a gauge.

Reply:
You're right, it's always resetting. Makes sense.

Reply:
Changed the counter name to totalAllocatedThreads to better reflect our intentions.